pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -191,18 +192,25 @@ def _query_job_progress(
         seq_builder = Sequence.from_abstract_repr(batch.sequence_builder)
         reg = seq_builder.get_register(include_mappable=True)
         all_qubit_ids = reg.qubit_ids
-        meas_basis = seq_builder.get_measurement_basis()
 
         results: dict[str, tuple[JobStatus, Result | None]] = {}
 
         for job in batch.ordered_jobs:
-            vars = job.variables
+            vars = job.variables or {}
+            meas_basis = (
+                seq_builder.build(**vars) if vars else seq_builder
+            ).get_measurement_basis()
             size: int | None = None
             if vars and "qubits" in vars:
                 size = len(vars["qubits"])
             if job.result is None:
                 results[job.id] = (JobStatus[job.status], None)
+                continue
```

Comment:
If the user doesn't define bitstrings observables, job.result will be None. In that case, don't we want instead to try to return the full_results as below ?`(JobStatus[job.status], Results.from_abstract_repr(job.full_result["serialised_results"]))`